[docs] Added section on using Helm chart to deploy Enterprise binaries (#5454)
hashicorp/consul: website/source/docs/platform/k8s/helm.html.md

      The name of the private key for the certificate file within the
      `secretName` secret.

## Using the Helm Chart to deploy Consul Enterprise

> This Helm chart can also be used to deploy Consul Enterprise by following a few extra steps.

Reviewer: ```suggestion
You can also use this Helm chart to deploy Consul Enterprise by following a few extra steps.
```

> Find the license file that was sent to you with your welcome email. It should have the extension `.hclic`. You will use the contents of this file to create a Kubernetes secret before installing the Helm chart.

Reviewer: ```suggestion
Find the license file that you received in your welcome email. It should have the extension `.hclic`. You will use the contents of this file to create a Kubernetes secret before installing the Helm chart.
```

Author: Accepted!

> -> **Note:**: If you cannot find your `.hclic` file, please contact your sales team or Technical Account Manager.

Reviewer: ```suggestion
-> **Note:** If you cannot find your `.hclic` file, please contact your sales team or Technical Account Manager.
```

You can use the following commands to create the secret:

```bash
secret=$(cat 1931d1f4-bdfd-6881-f3f5-19349374841f.hclic)
kubectl create secret generic consul-ent-license --from-literal="key=${secret}"
```

In your `values.yaml`, change the value of `global.image` to one of the enterprise [release tags](https://hub.docker.com/r/hashicorp/consul-enterprise/tags).

```yaml
global:
  image: "hashicorp/consul-enterprise:1.4.3-ent"
```

Add the name of the secret you just created to `server.enterpriseLicense`.

```yaml
server:
  enterpriseLicense:
    secretName: "consul-ent-license"
    secretKey: "key"
```

Add the `--wait` option to your `helm install` command. This will force Helm to wait for all the pods

> to become ready, then apply the license to your Consul cluster.

Reviewer: ```suggestion
to become ready before it applies the license to your Consul cluster.
```
